PermissionsScrapper/PermissionsScrapper/PermsDescription.cs

```diff
@@ -0,0 +1,141 @@
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace PermissionsScrapper
+{
+    public static class PermsDescription
+    {
+        static string _scopesDescriptionsJson = null;
+
+        /// <summary>
+        /// Timer function that fetches permissions descriptions from a Service Principal
+        /// and uploads them to a GitHub repo at specified times.
+        /// </summary>
+        /// <param name="myTimer">Trigger function every weekday at 9 AM UTC</param>
+        /// <param name="log"></param>
+        [FunctionName("DescriptionsScrapper")]
+        public static void Run([TimerTrigger("0 0 9 * * 1-5")]TimerInfo myTimer, ILogger log)
+        {
+            try
+            {
+                ApplicationConfig config = ApplicationConfig.ReadFromJsonFile("local.settings.json");
+
+                var result = GetAuthentication(config);
+
+                if (result != null)
+                {
+                    var scopesDescriptionsJson = GetScopesDescriptionsJson(config, result);
+
+                    if (!scopesDescriptionsJson.Equals(_scopesDescriptionsJson, StringComparison.OrdinalIgnoreCase))
```

Comment:
so every time the service restarts we'll end up with a Github write of the same information we already had?

Reply:
are you also assuming that we go with an always on plan here?  I'd think we'd want a consumption plan here as this is not a high frequency item.

Reply:
A call to GitHub to fetch the respective files from the main branch will be made on service start. A separate branch/PR will integrate GitHub into this project.